flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/StateTransition.java

```diff
@@ -55,6 +65,7 @@ public boolean equals(Object obj) {
 			StateTransition<T> other = (StateTransition<T>) obj;
 
```

Comment:
You have updated the `equals`, you should also update the `hashCode`  and the `toString` accordingly.